feat: Add WP escapers via Twig filters

Levdbas opened the pull request:
Related:

- #1848
- #1787

## Issue
Looking at the old issues I found this particular one. With escaping getting ever more important I wanted to take a shot at this.
Escaping currently in Timber is a bit cumbersome since you have to do things like this:

```twig
{{ my_content|e('esc_attr') }}
```
or even twig when setting the default Twig escaper already on in the environment. The |raw filter makes sure that the string is not double escaped.

```twig
{{ my_content|e('wp_kses_post')|raw }}
```

## Solution
Make escaping functions available via filters. 

- in `get_timber_escaper_filters` I add most of the default escapers that come with WP
- in `add_timber_escaper_filters` I have a simular function as `add_timber_filters` but add the following options by default:
```php
options' => [
  " 'is_safe' => ['html'],
],
```

## Impact
A better way of applying escapers

## Usage Changes
So instead of running `{{ my_content|e('esc_attr') }}` we can now run {{ my_content|esc_attr }} and it does not matter if the global escaping is on or off.

## Considerations
I did not spend much time on the docs at this time, but this is needed. We also might want to deprecate the old way of adding filters in the future, so this would be the recommended way.

If we want to implement this, we also need to take a look at  #1787 on updating the existing escaping examples in the PHP files.

## Testing
Yes. there were no escaper tests at this time so I added them and run some of them with global escaping on to test #1848

Levdbas committed:
feat: Add WP escapers via Twig filters

gchtr commented on src/Twig.php:
```diff
@@ -427,6 +427,73 @@ public function get_timber_filters()
+        ];
+
+        /**
+         * Filters the filters that are added to Twig.
```
```suggestion
         * Filters the escaping filters that are added to Twig.
```

gchtr reviewed (commented):
Oh that’s great, @Levdbas. That’s a good step towards making escaping easier.

> I did not spend much time on the docs at this time, but this is needed.

What ideas did you have to improve the docs, apart from what’s stated in #1787?

> We also might want to deprecate the old way of adding filters in the future, so this would be the recommended way.

Definitely something we could do.

Levdbas committed:
docs: update escaping examples

Levdbas committed:
docs: update inline escaping examples

Levdbas committed:
fix: Remove esc_url_raw since it is not recommended to use it on the front-end anyway.

Levdbas commented:
@gchtr , I have updated te escaping guide with new examples and removed a escaping function that was not recommended for being used on the front-end. About the deprecation of the existing escapers, do we want to do that right away?

We could do it like this for all escaper functions/filters

```php
        $esc_url = function (Environment $env, $string) {
            Helper::deprecated(
                "{{ url|e('esc_url') }}",
                "{{ url|esc_url }}",
                '2.1.0'
            );
            return \esc_url($string);
        };
```

[over here ](https://github.com/timber/timber/blob/6ebe3ef4eb4c23c05d6caf193d63cfdef972de38/src/Twig.php#L541-L548)

gchtr commented on src/Twig.php:
```diff
@@ -427,6 +427,70 @@ public function get_timber_filters()
+    public function get_timber_escaper_filters()
+    {
+        $escaper_filters = [
+            /* image filters */
```
```suggestion
```

gchtr reviewed (dismissed):
This looks good to me!

> About the deprecation of the existing escapers, do we want to do that right away?

No, this seems like a breaking change to me that would be something for 3.0.

Levdbas committed:
Update src/Twig.php

Co-authored-by: Lukas Gächter <lukas.gaechter@mind.ch>
